src/Prescription/Prescription.controller.ts

```diff
@@ -0,0 +1,204 @@
+import {
+  Controller,
+  Get,
+  UsePipes,
+  Query,
+  ValidationPipe,
+  Post,
+  Body,
+  Put,
+  Param,
+  InternalServerErrorException,
+  UseGuards,
+} from '@nestjs/common';
+import { CreatePrescriptionDto } from './dto/createPrescription.dto';
+import { GetPrescriptionDto } from './dto/getPrescription.dto';
+import { PrescriptionService } from './Prescription.service';
+import {
+  ApiOperation,
+  ApiResponse,
+  ApiTags,
+  ApiQuery,
+  ApiParam,
+} from '@nestjs/swagger';
+import * as _ from 'lodash';
+import { PrescriptionGuard } from '../Guards/Prescription.guard';
+
+// import { FhirService } from 'src/Integerations/fhirService';
+
+@ApiTags('prescriptions')
+@Controller('prescription')
+@UseGuards(PrescriptionGuard)
+export class PrescriptionController {
+  constructor(
+    private readonly prescriptionService: PrescriptionService, // private readonly fhirService: FhirService  //fhirService to Interact with External API
+  ) {}
+
+  @Get('search')
+  @ApiQuery({
+    name: 'page',
+    required: false,
+    type: Number,
+    description: 'Page number',
+  })
+  @ApiQuery({
+    name: 'size',
+    required: false,
+    type: Number,
+    description: 'Number of items to return',
+  })
+  @ApiQuery({
+    name: 'nhi',
+    required: true,
+    type: String,
+    description: 'NHI number for which we want prescription records',
+  })
+  @ApiOperation({ summary: 'Get prescriptions by NHI' })
+  @ApiResponse({
+    status: 200,
+    description: 'The found record',
+    type: [CreatePrescriptionDto],
+  })
+  @UsePipes(ValidationPipe)
+
```

Comment:
I highly doubt this space should be allowed here, it might not apply decorator on the controller